### PR TITLE
Add default param handling, Remove Spring dependency

### PR DESCRIPTION
**Pull request by Signavio-math**
Adds the following features
* Add new Only semantic next to Exactly semantic
  * default: asserts that configured annotations and params are present. Default values for not configured params are considered.
  * only: assert that only the configured annotations are present. Default values for not configured params are considered.
  * exactly: assert that only the configured annotations and params are present. No default values are considered.
* Removed all Spring dependencies from production code. The dependencies are still present in test scope. All reflection based 
* proper evaluation of AliasFor annotated fields/methods

### CODE REVIEW

File: src/main/java/de/tolina/common/validation/AnnotationValidation.java [hunk 1]
```diff
@@ -111,87 +140,132 @@ public void forMethod(@Nonnull final Method annotatedMethod) {
...
+						if (methodResult instanceof String) {
+							softly.assertThat((String) methodResult).as(description, declaredMethod.getName())
+									.isNullOrEmpty();
+						} else {
+							softly.assertThat(methodResult).as(description, declaredMethod.getName()).isNull();
+						}
+					}
 				}
+			} catch (IllegalAccessException | InvocationTargetException e) {
+				e.printStackTrace();
```
Comment (ghost) [inline src/main/java/de/tolina/common/validation/AnnotationValidation.java]: Please handle Exceptions.

File: src/main/java/de/tolina/common/validation/AnnotationValidation.java [hunk 2]
```diff
@@ -202,124 +276,219 @@ private boolean equalParamTypes(@Nonnull final Class<?>[] typesOne, @Nonnull fin
...
+			if (aliasForAnnotation.isPresent()) {
+				try {
+					aliasMethod = getAliasMethod(annotation, aliasForAnnotation.get());
+				} catch (NoSuchMethodException e) {
+					softly.assertThat(actualMethod)
+							.as("Referenced alias method %s not found.", aliasForAnnotation.get())
+							.isNotNull();
+					continue;
+				} catch (IllegalAccessException | InvocationTargetException e) {
+					e.printStackTrace();
```
Comment (ghost) [inline src/main/java/de/tolina/common/validation/AnnotationValidation.java]: Please handle Exceptions.

File: src/main/java/de/tolina/common/validation/AnnotationValidation.java [hunk 3]
```diff
@@ -202,124 +276,219 @@ private boolean equalParamTypes(@Nonnull final Class<?>[] typesOne, @Nonnull fin
...
+			final Object actualMethodResult;
+			final Object aliasMethodResult;
+			Object assertableResult = null;
+			try {
+				actualMethodResult = actualMethod.invoke(annotation);
+				assertableResult = actualMethodResult;
+				
+				assertMethodResult(actualMethodResult, expectedValues);
+			} catch (IllegalAccessException | InvocationTargetException e) {
+				e.printStackTrace();
```
Comment (ghost) [inline src/main/java/de/tolina/common/validation/AnnotationValidation.java]: Please handle Exceptions.

File: src/main/java/de/tolina/common/validation/AnnotationValidation.java [hunk 4]
```diff
@@ -202,124 +276,219 @@ private boolean equalParamTypes(@Nonnull final Class<?>[] typesOne, @Nonnull fin
...
+			} catch (AssertionError e) {
+				if (aliasForAnnotation.isPresent()) {
+					try {
+						aliasMethodResult = aliasMethod.invoke(annotation);
+						
+						assertMethodResult(aliasMethodResult, expectedValues);
+						
+						assertableResult = aliasMethodResult;
+					} catch (IllegalAccessException | InvocationTargetException e1) {
+						e1.printStackTrace();
```
Comment (ghost) [inline src/main/java/de/tolina/common/validation/AnnotationValidation.java]: Please handle Exceptions.

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by Signavio-math — COMMENTED** [review]
reflection exceptions now lead to failing tests

**Review by ghost — APPROVED** [review]

**Pull request merged**